// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.chengyun.detector";
option java_outer_classname = "DetectorProto";
option objc_class_prefix = "chengyun";

package detector;

// The greeting service definition.
service Detector {
  // Sends a greeting
  rpc Predict (DetectorRequest) returns (stream DetectorReply) {}
}
// The request message containing the user's name.
message DetectorRequest {
  string file = 1;
  float thresh=2;
  float hier_thresh=3;
  string token=4;
}

// The response message containing the greetings
message DetectorReply {
  string name = 1;
  int32 top = 2;
  int32 left = 3;
  int32 right = 4;
  int32 bottom = 5;
  float rate= 6;
}
